inference-engine/src/mkldnn_plugin/nodes/mkldnn_shuffle_channels_node.cpp

```diff
@@ -0,0 +1,843 @@
+    if (group == 0 || dst_dims[axis] % group)
+        THROW_IE_EXCEPTION << "ShuffleChannels layer with name " << getName() <<
+                              " gets incorrect group parameter! Group parameter must evenly divide the channel dimension!";
+
+    shuffle_size = dst_dims[axis];
+    group_size = dst_dims[axis] / group;
+}
+
+void MKLDNNShuffleChannelsNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    static const Precision supportedPrecision[] = {
+        Precision::FP32,
+        Precision::BF16,
+        Precision::I32,
+        Precision::I8,
+        Precision::U8
+    };
+
+    Precision inputPrecision = getCnnLayer()->insData[0].lock()->getPrecision();
+    Precision outputPrecision = getCnnLayer()->outData[0]->getPrecision();
+    if (outputPrecision == Precision::BF16 && !mayiuse(avx512_core))
+        outputPrecision = Precision::FP32;
+
+    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(inputPrecision);
+    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(outputPrecision);
+
+    // shuffle shouldn't change the precision itself
+    if (inputDataType != outputDataType)
+        inputDataType = outputDataType;
+
+    jit_mode = mayiuse(cpu::sse42) && std::find(std::begin(supportedPrecision), std::end(supportedPrecision), inputPrecision)
+                                   != std::end(supportedPrecision);
+
+    data_type = inputDataType;
+    data_size = MKLDNNExtensionUtils::sizeOfDataType(data_type);
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = false;
+    config.inConfs.resize(1);
+    config.outConfs.resize(1);
+    config.inConfs[0].constant = false;
+    config.outConfs[0].constant = false;
+    config.inConfs[0].inPlace = -1;
+    config.outConfs[0].inPlace = -1;
+
+    auto pushDesc = [&](memory::format inFormat, memory::format outFormat, memory::data_type dataType, impl_desc_type impl_type) {
+        config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), dataType, inFormat);
+        config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), dataType, outFormat);
+        supportedPrimitiveDescriptors.push_back({config, impl_type, outFormat});
+    };
+
+    if (jit_mode) {
+        impl_desc_type impl_type = impl_desc_type::jit_sse42;
+        if (mayiuse(cpu::avx512_common)) {
+            impl_type = impl_desc_type::jit_avx512;
+        } else if (mayiuse(cpu::avx2)) {
+            impl_type = impl_desc_type::jit_avx2;
+        }
+
+        pushDesc(MKLDNNMemory::GetPlainFormat(memory::dims(getParentEdgeAt(0)->getDims().ndims())),
+             MKLDNNMemory::GetPlainFormat(memory::dims(getChildEdgeAt(0)->getDims().ndims())), data_type, impl_type);
+        // Current tests show that reorder+planar-shuffleChannels outperforms blocked-shuffleChannels, given axis == 1.
+        // It makes sense to support blocked/byChannel layout only when axis != 1.
```

Comment:
In real models majority of the cases involve axis=1. That's why we should consider support for blocked/byChannel layouts as requirement. We have to find how to implement this in more efficient way. I am referring to my comment from the previous review iterations: https://github.com/openvinotoolkit/openvino/pull/1360#discussion_r531393184. Please take a look on these instructions - they should significantly outperform "gather" in most cases. Moreover gather supports only 4 bytes types and for lower precisions you are emulating it by loading via scalar instructions - which will be dramatically slow for bf16 and int8 models.